obfuscator.go

```diff
@@ -0,0 +1,76 @@
+package sqllexer
+
+import (
+	"regexp"
+	"strings"
+)
+
+type SQLObfuscatorConfig struct {
+	ReplaceDigits    bool
+	DollarQuotedFunc bool
+}
+
+type SQLObfuscator struct {
+	config *SQLObfuscatorConfig
+}
+
+func NewSQLObfuscator(config *SQLObfuscatorConfig) *SQLObfuscator {
+	return &SQLObfuscator{config: config}
+}
+
+const (
+	StringPlaceholder = "?"
+	NumberPlaceholder = "?"
+)
+
+// Obfuscate takes an input SQL string and returns an obfuscated SQL string.
+// The obfuscator replaces all literal values with a single placeholder
+func (o *SQLObfuscator) Obfuscate(input string) string {
+	var obfuscatedSQL string
+
+	lexer := NewSQLLexer(input)
+	for token := range lexer.ScanAllTokens() {
```

Comment:
So this will allow us to scan all tokens as we see them on the channel? I think at some point we are going to want to support a stateful type parser, do you think we can easily extend this to support that?

Reply:
In this case the `ScanAll` func can be used to return an array of tokens. Or we can consider make `Token` a double linked list with pointers to previous and next token.